src/ddmd/func.d

```diff
@@ -156,6 +156,61 @@ enum FUNCFLAGinlineScanned    = 0x20;   /// function has been scanned for inline
 enum FUNCFLAGinferScope       = 0x40;   /// infer 'scope' for parameters
 
 
+/****************************************************
+ * Determine whether an 'in' contract is declared inside
+ * the given function or any of its overrides.
+ * Params:
+ *      fd = the function to search
+ * Returns:
+ *      true    found an 'in' contract
+ *      false   didn't find one
+ */
+private bool needsFrequire(FuncDeclaration fd)
+{
+    if (fd.frequire)
+        return true;
+
+    for (size_t i = 0; i < fd.foverrides.dim; i++)
+    {
+        FuncDeclaration fdv = fd.foverrides[i];
+
+        if (fdv.frequire)
+            return true;
+
+        if (needsFrequire(fdv))
+            return true;
+    }
+    return false;
+}
+
+/****************************************************
+ * Determine whether an 'out' contract is declared inside
+ * the given function or any of its overrides.
```

Comment:
Why is `or any of its overrides` required? It wasn't before.

Reply:
I might have misread the implementation of `mergeFensure` I was under the impression that these merge routines *could* create a new pointer where `fensure` was previously `null`. The loop and recursion logic was copied from those routines.

Reply:
Actually, you are right about `needsFrequire`.  If the function being analysed does not have an `in` contract, then nothing will be generated anyway.  So undoing that from this PR.